### PR TITLE
Setup Github workflow for IPC provider & Ipc few requests at the same time returns in one promise

### PR DESCRIPTION
**Pull request by avkos**
## Description

#5147 and #5185

Please include a summary of the changes and be sure to follow our [Contribution Guidelines](../blob/1.x/CONTRIBUTIONS.md).

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` with success.
- [ ] I have tested the built `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.

### CODE REVIEW

File: packages/web3-utils/src/chunk_response_parser.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+import { JsonRpcResponse } from 'web3-types';
+import { InvalidResponseError } from 'web3-errors';
+
+export class ChunkResponseParser {
```
Comment (jdevcs) [inline packages/web3-utils/src/chunk_response_parser.ts:20]: Good work

**Review by jdevcs — APPROVED** [review]

File: packages/web3-utils/src/chunk_response_parser.ts (hunk 1 above)
Comment (jdevcs) [inline packages/web3-utils/src/chunk_response_parser.ts:20]: Created https://github.com/ChainSafe/web3.js/issues/5349 for more tests we can do in another PR.

**Review by luu-alex — APPROVED** [review]
